brownie/project/main.py

```diff
@@ -223,7 +224,37 @@ def load(self) -> None:
         self._active = True
         _loaded_projects.append(self)
 
-    def _get_changed_contracts(self, compiled_hashes: Dict) -> Dict:
+    def _flatten_contract(self, build_project: Build) -> str:
+        flatterner = "flatten_contracts.sol"
+        flatten_path = self._path.joinpath(flatterner)
+
+        base_contract_name = ""
+        has_dependencies = 0
+        for contract_name, _ in build_project.items():
+            contract_dependencies_length = len(build_project.get_dependents(contract_name))
+            if contract_dependencies_length == has_dependencies:
+                base_contract_name = contract_name
+        base_contract = build_project._sources._contracts[base_contract_name]
+
+        for contract in build_project._sources._source.keys():
+            if contract == base_contract:
+                continue
+
+            with flatten_path.open("a") as fp:
```

Comment:
It would be more efficient to prepare the data in a string and write to the file system once, instead of opening and appending to the same file many times.

Also, using append means the 2nd time this runs I will have two copies of the flattened contract(s) in the same file.